### PR TITLE
Add windows support

### PR DESCRIPTION
**Pull request by Siebjee**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # bamboo_agent
 
 ## Build Status
-[![Build Status](https://travis-ci.org/knuedge/bamboo_agent.svg?branch=master)](https://travis-ci.org/knuedge/bamboo_agent)
+[![Build Status](https://travis-ci.org/Siebjee/bamboo_agent.svg?branch=master)](https://travis-ci.org/Siebjee/bamboo_agent)
```
Comment (jgnagy) [inline README.md]: Please remove this from this branch.

Comment (Siebjee) [reply README.md]: Pushed the revert to the branch.

**Comment by jgnagy** [comment]
I believe the reason all the tests now fail in Travis is related to your lookup to `$facts['os']` which isn't set when rspec runs.

Can you please do the following:

* Modify the spec files to integrate the `${os}` into the `:facts` configuration. Something like `let(:facts) { facts.merge({:os => { :family => 'Debian' } }) }` (obviously integrating other facts data set in the tests and set appropriately for the actual OS being tested).
* Make sure you have working tests for Windows with the `:facts` `['os']['family']` set appropriately.

I believe that's all that is required to get the tests working again.

File: examples/crossplatform.pp [hunk 2]
```diff
@@ -0,0 +1,89 @@
+case $facts['os']['family'] {
```
Comment (dschaaff) [inline examples/crossplatform.pp]: I don't have a puppet dev environment setup right to test, but I believe this should be `facts[:osfamily]` For example

```
describe 'foo' do
  on_supported_os.each do |os, facts|
    context "on #{os}" do
      let(:facts) do
        facts
      end
      if facts[:osfamily] == 'Debian' then
        it { is_expected.to compile }
      end
    end
  end
end
```

Comment (dschaaff) [reply examples/crossplatform.pp]: Yep, see the docs for the puppet facts gem these tests use https://github.com/mcanevet/rspec-puppet-facts and their example of a case statement
```
# If you need any to specify any operating system specific tests
      case facts[:osfamily]
      when 'Debian'
        ...
      else
        ...
      end
```

Comment (jgnagy) [reply examples/crossplatform.pp]: Newer versions of Facter do support what they call "Modern Facts" as part of core:

https://docs.puppet.com/facter/3.5/core_facts.html#os

That said, I agree that using `facts[:osfamily]` would be better to retain backwards compatibility.

Comment (dschaaff) [reply examples/crossplatform.pp]: I thought I was looking at a spec file haha. I blame my iPhone ;)

**Comment by Siebjee** [comment]
Updated the spec files, travis is still failing to build :( .

**Comment by jgnagy** [comment]
You appear to have syntax errors in your rspec changes:

```ruby
let (:facts) do {
```

This should be either

```ruby
let(:facts) do
...
end
```

Or

```ruby
let(:facts) {
...
}
```

This is how Ruby handles blocks.

Given the difficulty with fixing the tests, I will make a "development" branch, and you can recreate this pull request to merge into that branch rather than `master` if you're certain the Windows portion of your code changes works. This way we can accept your PR into a branch where we can safely assist with assessing and fixing the rspec changes.

**Pull request closed without merging**